dbt_semantic_interfaces/validations/saved_query.py

```diff
@@ -29,14 +31,30 @@ class SavedQueryRule(SemanticManifestValidationRule[SemanticManifestT], Generic[
     """
 
     @staticmethod
-    def _model_requires_primary_entity(semantic_model: SemanticModel) -> bool:
-        return len(semantic_model.dimensions) > 0
+    @validate_safely("Validate the group-by field in a saved query.")
+    def _check_group_bys(valid_group_by_element_names: Set[str], saved_query: SavedQuery) -> Sequence[ValidationIssue]:
+        issues: List[ValidationIssue] = []
+
+        for group_by_item_name in saved_query.group_bys:
+            structured_name = DunderedNameFormatter.parse_name(group_by_item_name)
```

Comment:
We already talked about it in a call, but documenting it here. This assumes that the `group_bys` will be strings of the form `'<primary_enitity>__<dimension>'`. Although eventually we want to support that, the strings right now will be of the form `'Dimension("<primary_entity>__<dimension>")'`. We should probably have a `GroupByParser`, similar to the `WhereFilterParser` for parsing group_bys, and use that for validating that the group_bys have valid structuring.

Reply:
Re-used the `WhereFilterParser` for now - there's some work going on right now to support `Dimension(...).grain(...)`, so I'll wait until that's in place to switch out the implementation.